evap/results/views.py

```diff
@@ -376,38 +399,51 @@ def evaluation_detail_parse_get_parameters(request, evaluation):
     if not evaluation.can_results_page_be_seen_by(request.user):
         raise PermissionDenied
 
-    view = request.GET.get("view", "public" if request.user.is_reviewer else "full")
-    if view not in ["public", "full", "export"]:
-        view = "public"
+    view_general_results = request.GET.get(
+        "view_general_results",
+        "full",
+    )
+    if view_general_results not in ["full", "ratings"]:
+        view_general_results = "full"
+
+    view_contributor_results = request.GET.get(
+        "view_contributor_results",
+        "full",
+    )
+    if view_contributor_results not in ["full", "ratings", "personal"]:
+        view_contributor_results = "full"
 
     view_as_user = request.user
     try:
         contributor = get_object_or_404(UserProfile, pk=request.GET.get("contributor_id", request.user.id))
     except ValueError as e:
         raise BadRequest from e
 
-    if view == "export" and request.user.is_staff:
+    if view_contributor_results == "personal" and request.user.is_staff:
         view_as_user = contributor
     contributor_id = contributor.pk if contributor != request.user else None
 
     represented_users = [view_as_user]
-    if view != "export":
-        represented_users += list(view_as_user.represented_users.all())
-    # redirect to non-public view if there is none because the results have not been published
-    if not evaluation.can_publish_rating_results and view == "public":
-        view = "full"
+    represented_users += list(view_as_user.represented_users.all())
 
-    return view, view_as_user, represented_users, contributor_id
+    return view_general_results, view_contributor_results, view_as_user, represented_users, contributor_id
```

Comment:
Can we use strong typing (optimally statically checkable) for these? (-> have an enum representing the possible values for `view_general_results`, `view_contributor_results`, instead of using strings)